WPMedia/arc-themes-components
# Tmedia 658 component stack

### Update Feb 24:  Implement @matthewroach's children iteration suggestion (Amazing resource!) as well as ensuring divider are only implemented in between elements.  Added a few more stories as well.  @nschubach Had a couple concerns:

1. The use of `additionalClassNames` is a bit messy and error prone.  I kind of agree on this.  However, this is a bit of a pattern in this framework and we need to discuss as a team what we want to do.
2. Nick suggests that I am exposing Flex concepts too much and should be abstract more.  My thoughts are that by making this more "Flex" oriented to allow users to pick this up faster.  But I am open for more discussion!!


### Update Feb 23:  I have added some additional properties: Justification _and_ alignment (to allow for more Flex behaviors) and a divider that uses an `<hr />` per @matthewroach suggestion.  The issue I have with using an `<hr>` over using absolute positioned pseudo elements for a divider is that with the `<hr>`, you need to iterate over the children elements and that results in a key violation.  I have added keys to the children element in the tests and put a note in it in the documentation.  I am taking this off WIP and putting it up for formal review.

### Update Feb 22:  I have added a flex wrap facility to this component.  I have also made modifications to Matthew's requests below.  Still WIP status as I am not how to properly implement a divider (because of the use of gap)...this is feeling like a separate component.

### Note still in WIP.  Need clarification on what is meant by adding a divider between child elements and if we are ready to define that yet.  Also the last storybook story uses a token that isn't defined yet.

## Description
Ingrates the Stack component into the repository. 

## Jira Ticket
[[- [TMEDIA-](https://arcpublishing.atlassian.net/browse/TMEDIA-)](https://arcpublishing.atlassian.net/browse/TMEDIA-651)](https://arcpublishing.atlassian.net/browse/TMEDIA-658)

## Acceptance Criteria

- Provide a <Stack> component that can
- layout children vertically
- layout children horizontally
- layout children in different alignments: start, center, end
- have a divider between child elements
- Have Scss file that adds the hooks to the Sass library
- Have test coverage
- Have storybook documentation

## Test Steps

1. Checkout this branch `TMEDIA-658-component-stack`
4. Run storybook `npm run storybook` and audit the stack documentation and stories
5. Run tests `npm run test` and ensure all tests run.
